src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.Method.cs

```diff
@@ -32,9 +32,35 @@ public void newline_is_legal_at_the_beginning_of_a_request()
                 GET https://example.com
                 """);
 
-            result.SyntaxTree.RootNode
-                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
-                  .MethodNode.ChildTokens.First().Kind.Should().Be(HttpTokenKind.NewLine);
+            var methodNode = result.SyntaxTree
+                                   .RootNode
+                                   .ChildNodes
+                                   .Should()
+                                   .ContainSingle<HttpRequestNode>().Which
+                                   .MethodNode;
+
+            methodNode.ChildTokens.First().Kind.Should().Be(HttpTokenKind.NewLine);
+            methodNode.Text.Should().Be("GET");
+        }
+
+        [Fact]
+        public void comment_is_legal_at_the_beginning_of_a_request()
+        {
+            var result = Parse(
+                """
+                # this is a comment
+                GET https://example.com
+                """);
+
+            var methodNode = result.SyntaxTree
+                                   .RootNode
+                                   .ChildNodes
+                                   .Should()
+                                   .ContainSingle<HttpRequestNode>().Which
+                                   .MethodNode;
+
+            methodNode.ChildTokens.First().Kind.Should().Be(HttpTokenKind.NewLine);
```

Comment:
This assert seems less relevant for this test (and probably duplicates coverage from the previous test on line 42). Perhaps the assert should be changed to check for the comment node instead?

```suggestion
            methodNode.ChildNodes.First().Should().BeOfType(typeof(HttpCommentNode));
```

Reply:
Nice catch.